feat(module:tree-view): add tree-view component (#6161)
NG-ZORRO/ng-zorro-antd: components/tree-view/doc/index.en-US.md

| [nzMaxBufferPx] | The number of pixels worth of buffer to render for when rendering new nodes | `number` | `28 * 10` |

### [nzTreeNodeDef]

Directive to define `nz-tree-node`.

| Property | Description | Type | Default |
| --- | --- | --- | --- |
| [nzTreeNodeDefWhen] | Function that should return true if this node template should be used for the provided node data and index. If left undefined, this node will be considered the default node template to use when no other when functions return true for the data. For every node, there must be at least one when function that passes or an undefined to default. | `(index: number, nodeData: T) => boolean` | - |

### nz-tree-node

The tree node container component, which needs to be defined by the `nzTreeNodeDef` directive.
### [nzTreeNodePadding]

```html
<nz-tree-node nzTreeNodePadding></nz-tree-node>
```

Show node indentation by adding `padding` **Best Performance**.

### nzTreeNodeIndentLine

```html
<nz-tree-node nzTreeNodeIndentLine></nz-tree-node>
```

Show node indentation by adding indent lines.

### nz-tree-node-toggle

> Toggle part, used to expand / collapse the node.

Reviewer: ```suggestion
A toggle which is used to expand / collapse the node.
```

| Property | Description | Type | Default |
| --- | --- | --- | --- |
| [nzTreeNodeToggleRecursive] | Is it recursively expand / collapse | `boolean` | `false` |

### nz-tree-node-toggle[nzTreeNodeNoopToggle]

> Does not do anything to the toggle section, available with a placeholder or display icons.

Reviewer: ```suggestion
A toggle that does no actions. This can be used for placeholders or displays icons.
```

### [nz-icon][nzTreeNodeToggleRotateIcon]

> Define icons in the toggle part, which will automatically rotate with the collapse/expand state.

Reviewer: ```suggestion
Define an icon in the toggle, which it will automatically rotate depending on the collapse/expand state.
```

### [nz-icon][nzTreeNodeToggleActiveIcon]

> Defines the icons in the toggle section to have an active style, which can be used for loading icon.

Reviewer: ```suggestion
Defines an icon in the toggle for an active style, which it  can be used for the loading state.
```

### nz-tree-node-option

> Defines the selectable part of a node.

Reviewer: ```suggestion
Defines the selectable feature of a node.
```

| Property | Description | Type | Default |
| --- | --- | --- | --- |
| [nzSelected] | Whether the option is selected | `boolean` | `false` |
| [nzDisabled] | Whether the option is disabled | `boolean` | `false` |
| (nzClick) | Event on click | `EventEmitter<MouseEvent>` | - |

### nz-tree-node-checkbox

> Defines the checkable parts of a node.

Reviewer: ```suggestion
Defines the checkbox feature of a node.
```
